### PR TITLE
Recipe api client uses common errors package.

### PR DESCRIPTION
**Pull request by DavidSubiros**
### What

- recipe-api client uses the common error package
- created a Checker() function for cantabular client

### How to review

- Make sure code changes make sense
- Make sure unit tests pass

### Who can review

anyone

### CODE REVIEW

**Review by redhug1 — DISMISSED** [review]
The changes look sensible.
The tests, audit and go mod tidy do not show any issues, so:
LGTM

File: recipe/recipe.go [hunk 1]
```diff
@@ -16,10 +17,10 @@ func (c *Client) GetRecipe(ctx context.Context, userAuthToken, serviceAuthToken,
...
 	res, err := c.doGetWithAuthHeaders(ctx, userAuthToken, serviceAuthToken, uri)
 	if err != nil {
-		return nil, &Error{
-			err:        fmt.Errorf("failed to get response from Recipe API: %s", err),
-			statusCode: http.StatusInternalServerError,
-		}
+		return nil, dperrors.New(
+			fmt.Errorf("failed to get response from Recipe API: %s", err),
+			http.StatusInternalServerError,
+			nil)
```
Comment (ChrysmOre) [inline recipe/recipe.go:23]: Bracket should be closed on the next line

File: recipe/recipe.go [hunk 2]
```diff
@@ -32,27 +33,21 @@ func (c *Client) GetRecipe(ctx context.Context, userAuthToken, serviceAuthToken,
...
-			err:        fmt.Errorf("failed to read response body: %s", err),
-			statusCode: res.StatusCode,
-			logData: log.Data{
-				"response_body": string(b),
-			},
-		}
+		return nil, dperrors.New(
+			fmt.Errorf("failed to read response body: %s", err),
+			res.StatusCode,
+			log.Data{"response_body": string(b)})
```
Comment (ChrysmOre) [inline recipe/recipe.go:39]: bracket should be closed on the next line (keep in line with where scope was opened)

File: recipe/recipe.go [hunk 3]
```diff
@@ -32,27 +33,21 @@ func (c *Client) GetRecipe(ctx context.Context, userAuthToken, serviceAuthToken,
...
-			err:        fmt.Errorf("failed to unmarshal response body: %s", err),
-			statusCode: http.StatusInternalServerError,
-			logData: log.Data{
-				"response_body": string(b),
-			},
-		}
+		return nil, dperrors.New(
+			fmt.Errorf("failed to unmarshal response body: %s", err),
+			http.StatusInternalServerError,
+			log.Data{"response_body": string(b)})
```
Comment (ChrysmOre) [inline recipe/recipe.go:50]: bracket should be closed on the next line (keep in line with where scope was opened)

File: recipe/recipe_test.go [hunk 4]
```diff
@@ -68,11 +69,10 @@ func TestGetRecipe(t *testing.T) {
...
 			recipe, err := recipeClient.GetRecipe(ctx, testUserAuthToken, testServiceToken, recipeID)
-			So(err, ShouldResemble, &Error{
-				err:        errors.New("failed to unmarshal error response body: invalid character 'r' looking for beginning of value"),
-				statusCode: http.StatusBadRequest,
-				logData:    log.Data{"response_body": "[response body empty]"},
-			})
+			So(err, ShouldResemble, dperrors.New(
+				errors.New(""),
+				http.StatusBadRequest,
+				nil))
```
Comment (ChrysmOre) [inline recipe/recipe_test.go:75]: last bracket should be closed on the next line (keep in line with where scope was opened)

File: recipe/recipe_test.go [hunk 5]
```diff
@@ -85,10 +85,10 @@ func TestGetRecipe(t *testing.T) {
...
 		Convey("Then whe GetRecipe is called, one GET /recipes/ID call is performed and the expected error is returned", func() {
 			recipe, err := recipeClient.GetRecipe(ctx, testUserAuthToken, testServiceToken, recipeID)
-			So(err, ShouldResemble, &Error{
-				err:        errors.New("failed to get response from Recipe API: recipe API error"),
-				statusCode: http.StatusInternalServerError,
-			})
+			So(err, ShouldResemble, dperrors.New(
+				errors.New("failed to get response from Recipe API: recipe API error"),
+				http.StatusInternalServerError,
+				nil))
```
Comment (ChrysmOre) [inline recipe/recipe_test.go:91]: bracket should be closed on the next line (keep in line with where scope was opened)

File: recipe/recipe_test.go [hunk 6]
```diff
@@ -104,11 +104,10 @@ func TestGetRecipe(t *testing.T) {
...
 			recipe, err := recipeClient.GetRecipe(ctx, testUserAuthToken, testServiceToken, recipeID)
-			So(err, ShouldResemble, &Error{
-				err:        errors.New("failed to unmarshal response body: invalid character 'i' looking for beginning of value"),
-				statusCode: http.StatusInternalServerError,
-				logData:    log.Data{"response_body": "invalidRecipeBody"},
-			})
+			So(err, ShouldResemble, dperrors.New(
+				errors.New("failed to unmarshal response body: invalid character 'i' looking for beginning of value"),
+				http.StatusInternalServerError,
+				log.Data{"response_body": "invalidRecipeBody"}))
```
Comment (ChrysmOre) [inline recipe/recipe_test.go:110]: bracket should be closed on the next line (keep in line with where scope was opened)

**Review by ChrysmOre — APPROVED** [review]

**Pull request merged**